### ISSUE TITLE
[Bug]: "Not enough free space" message when trying to upload file from browser

### ISSUE THREAD

**Issue by billtzim** [nextcloud/server#33418]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Users have reported that they are unable to upload files using their web browsers (firefox,chrome,edge,chromium tested). This happens by using the + button and drag n drop. In all cases a notification appears that says "Not enough space left". 

![image](https://user-images.githubusercontent.com/5667643/182123286-a80afc1f-c482-49ea-ab79-9d51197ff05e.png)

The space left on the device is OK (11% filled from 1TB)

`/mnt/nfs-vol1/myclouddata  1.0T  105G  920G  11% /nextcloud_nfsdata
`

If i try to drag n drop a file on to a folder then the upload happens without a problem

Also, uploads / syncs from nextcloud client happens without problem (tested client on windows, ubuntu, android)

### Steps to reproduce

1. Go to a folder where you want to upload a file
2. Try to upload it from + button or drag n drop it in place
3. File is not uploaded, message "Not enough space left" appears

### Expected behavior

The file to be uploaded

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 22.2.3 to 22.2.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "mycloud.econ.uth.gr"
        ],
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "version": "24.0.3.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Athens",
        "log_rotate_size": "1048576000",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0,
        "updater.release.channel": "stable",
        "preview_office_cl_parameters": " --headless --nologo --nofirststartwizard --invisible --norestore --convert-to pdf --outdir ",
        "maintenance": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "lost_password_link": "***REMOVED SENSITIVE VALUE***",
        "filesystem_check_changes": 1,
        "mysql.utf8mb4": true,
        "app_install_overwrite": [
            "impersonate",
            "files_retention",
            "jsloader",
            "gallery",
            "files_antivirus",
            "nextbackup"
        ],
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false,
        "default_phone_region": "GR"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - admin_audit: 1.14.0
  - apporder: 0.15.0
  - bruteforcesettings: 2.4.0
  - calendar: 3.4.2
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.2.0
  - contactsinteraction: 1.5.0
  - dav: 1.22.0
  - external: 4.0.0
  - extract: 1.3.5
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_antivirus: 3.3.0
  - files_external: 1.16.1
  - files_pdfviewer: 2.5.0
  - files_retention: 1.13.2
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - groupfolders: 12.0.1
  - impersonate: 1.11.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - quota_warning: 1.14.0
  - ransomware_protection: 1.13.0
  - recommendations: 1.3.0
  - richdocuments: 6.1.1
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - spreed: 14.0.3
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_ldap: 1.14.1
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - dashboard: 4.0.6
  - encryption: 2.2.0
  - files_external_gdrive: 0.4.0
  - files_external_onedrive: 1.1.0
  - jsloader: 1.5.0
  - keeweb: 0.6.9
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
No related error messages are logged when trying to upload files and the message "Not enough space left" appears.
```

### Additional info

The problem didn't exist with version 24.0.2, first appeared after the upgrade to 24.0.3

The problem exists when i try to upload something either from the + button or by drag 'n drop to the destination folder

If i drag n drop the file to be uploaded on to the folder that i want to be uploaded (and not going in the folder) then the file is uploaded with no problem. Below the logs from such an upload

{"reqId":"Yuec8N9S4WPiMMK-9HxPqwAAABc","level":0,"time":"2022-08-01T12:29:20+03:00","remoteAddr":"10.64.109.236","user":"USERNAME","app":"no app in context","method":"PUT","url":"/remote.php/webdav/Mail-attachments/%CE%943.pdf","message":"mount at /USERNAME/ has no provider set, performing full setup","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:103.0) Gecko/20100101 Firefox/103.0","version":"24.0.3.2","data":[]}
{"reqId":"Yuec8N9S4WPiMMK-9HxPqwAAABc","level":0,"time":"2022-08-01T12:29:20+03:00","remoteAddr":"10.64.109.236","user":"USERNAME","app":"files_antivirus","method":"PUT","url":"/remote.php/webdav/Mail-attachments/%CE%943.pdf","message":"Response :: stream: OK\n","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:103.0) Gecko/20100101 Firefox/103.0","version":"24.0.3.2","data":{"app":"files_antivirus"}}

**Comment by umlaeute** [comment]
have you checked whether there's enough space on the tempdirectory?

IIRC, files will be first uploaded to the tempdirectory, and then moved to the final destination.
Search `config/config.sample.php` for the `tempdirectory` directive.

**Comment by billtzim** [comment]
Hello umlaeute

i don't use the tempdirectory directive as it's used to override usage of the systems default temp directory. The partition where the system's temp directory is located has plenty of free space (18GB)

Apart from that as i mentioned i can upload files normally if i drug n drop on a folder instead of doing it in the folder

**Comment by umlaeute** [comment]
> Apart from that as i mentioned i can upload files normally if i drug n drop on a folder instead of doing it in the folder

i'm not sure i can parse this... what's the difference between "drag'n'drop *on* a folder" resp. "drag'n'drop it in place" (aka "*in* the folder")?

**Comment by billtzim** [comment]
OK, let's say i want to upload a file in a folder with path for example Home\myfolder. I can upload the file by going on the Home directory, there exists the icon of myfolder and drag the file to be uploaded on the folder icon. The file is uploaded successfully and if i click the myfolder icon i will see all it's contents, including the newly uploaded file.
If, on the other case, go directly to the contents of the folder myfolder and drag n drop the file "in place" then i have the "not enough free space" message

Does this clears it up for you?

**Comment by stephenitpro** [comment]
I too am experiencing this after the recent ugprade to 24.0.3.  Running Nginx, I suspect the configuration for Nginx requires a tweek but unsure what to change to resolve this issue.  Both my data and temp folders have 10s of terabytes of free spaces. Oddly there are no errors showing up within the Nextcloud.log file except for the pop-up "not enough free space" top right of browser.

**Comment by stephenitpro** [comment]
Similar behaviour of drag-droping files into white space versus dropping them onto a subfolder.  , I cannot simply drop them or use the + upload bottom. Only uploads files IF I drag files to a folder, then it uploads fine. There is definitely some broken Agax or other reliant Nginx mechanism to upload that breaks.  Using Webdav, Nextcloud App, and Nextcloud Desktop Sync client works fine.

**Comment by stephenitpro** [comment]
In the nginx access logs, oddly seeing "GET /apps/files/ajax/getstoragestats?dir=%2F HTTP/2.0" 302 0 "-", using the provided latest nginx configuration

**Comment by stephenitpro** [comment]
alright, after further digging, I finally tracked it down to apcu and the opcache caches and file caches.  It was doing something silly and trying to reference code that was cached.  Funny, I had restarted php-fpm, redis, and nginx many times but there were lingering caches that were being disruptive.  Once I cleared the /var/lib/php/opache using this little cache tool provided by  https://gordalina.github.io/cachetool/ I was able to upload again. So for me, not a bug but rather a dirty cache caused the issue.

**Comment by billtzim** [comment]
I can also confirm that this is a cache problem. I also tried to restart / reboot everything but no luck. I did not use the tool mentioned above. I remembered thinking of using the occ maintenance:repair  option but never actually doing it. After running the command it cleared whatever cache had the issue and uploading is normal again.
@stephenitpro thanks for the hint and memory jog :)
Bug or not i am closing this issue since we have an answer and hope it helps someone in the future
